### ISSUE TITLE
sc2: SC2-compatible REINFORCE with multi-head spatial policy

### ISSUE THREAD

**Issue by espenhk** [espenhk/gamer-ai#107]
The existing `REINFORCEPolicy` (`framework/policies.py`) uses a single softmax head over 9 discrete actions. For SC2, the action has two independent components: **fn_idx** (which PySC2 function to call) and **(x, y)** spatial coordinates. Treating these as a single 9-element distribution collapses the spatial structure: the policy cannot independently vary its function choice and spatial target, which limits generalisation across screen positions.

## Architectural change — two-head policy network

```
obs (13 or 21 dims)
        │
   [FC layers]
        │
  ┌─────┴──────┐
  │            │
fn_logits   spatial_logits
(6 dims,    (9 cell grid,
 softmax)    softmax)
  │            │
fn_idx      cell_idx → (x, y)
```

Both heads are trained jointly via the REINFORCE update:
```
loss = -log π(fn_idx | obs) * G_t
     - log π(cell | obs) * G_t
```

## Available-actions masking

Before sampling `fn_idx`, zero out log-probs of unavailable functions using `info["available_fn_ids"]` (added in the SC2 NeuralDQN issue):

```python
fn_logits[unavailable_fn_mask] = -np.inf
fn_probs = softmax(fn_logits)
fn_idx = np.random.choice(n_fns, p=fn_probs)
```

The spatial head is always unmasked — all 9 cells are valid spatial targets.

## Episode-length considerations

SC2 episodes are long and rewards are sparse (score delta per step is small; win/loss only at terminal). REINFORCE accumulates returns over the full episode — variance is high. Mitigations:

- Use `baseline: "running_mean"` (already supported in `REINFORCEPolicy`)
- Increase `gamma` to `0.995` to preserve long-horizon credit assignment
- Add entropy regularisation (`entropy_coeff: 0.05`) to prevent premature collapse to `no_op`

## Recommended hyperparams (`games/sc2/config/training_params.yaml`)

```yaml
policy_type: sc2_reinforce
policy_params:
  hidden_sizes: [128, 64]
  learning_rate: 0.0003
  gamma: 0.995
  entropy_coeff: 0.05
  baseline: running_mean
```

## Tasks

- [ ] Implement `SC2REINFORCEPolicy` in `games/sc2/sc2_policies.py` with two-head MLP (fn_head + spatial_head)
- [ ] Accept `available_fn_ids` from info dict; mask fn_logits before sampling and before log-prob computation
- [ ] Accumulate separate log-prob trajectories for fn and spatial heads; sum losses at episode update
- [ ] Register as `sc2_reinforce` in the policy factory
- [ ] Tests: two-head forward pass shapes correct; masked fn sampling never selects unavailable action; gradients flow to both heads; entropy term reduces over training

## Relevant files

- `framework/policies.py` — `REINFORCEPolicy` base class
- `games/sc2/client.py` — `available_fn_ids` in info dict (see SC2 NeuralDQN issue)
- `games/sc2/actions.py` — grid cell → (x, y) mapping
- New: `games/sc2/sc2_policies.py`
